src/main/java/com/example/gymApp/service/TraineeService.java

```diff
@@ -1,141 +1,104 @@
-      }
-
-
-      if (!traineeDAO.createTrainee(trainee)) {
-        throw new NoSuchElementException("Trainee with ID " + id + " could not be created.");
-      }
-      log.info("Trainee {} has been created successfully.", id);
-      return true;
-
-    } catch (IllegalArgumentException e) {
-      log.error("Error creating trainee: {}", e.getMessage());
-      return false; // Возвращаем false, если произошла ошибка валидации
+  @Transactional
+  public Trainee createTrainee(String firstName, String lastName, String username, String password,
+      LocalDate dateOfBirth, String address) {
+    if (userRepository.findByUsername(username).isPresent()) {
+      throw new IllegalArgumentException("Username already exists.");
     }
-  }
-
 
 
-  public boolean updateTrainee(Long id, String firstName, String lastName, LocalDate dateOfBirth,
-      String address) {
-    try {
+    User user = new User();
+    user.setFirstName(firstName);
+    user.setLastName(lastName);
+    user.setUsername(username);
+    user.setPassword(password);
+    user.setActive(true);
 
-      Trainee trainee = getTraineeById(id);
+    // no need to save it separately! И
+//    userRepository.save(user);
 
-      trainee.setFirstName(firstName);
-      trainee.setLastName(lastName);
-      trainee.setDateOfBirth(dateOfBirth);
-      trainee.setAddress(address);
 
-      validateTrainee(trainee);
+    Trainee trainee = new Trainee();
+    trainee.setUser(user);
+    trainee.setDateOfBirth(dateOfBirth);
+    trainee.setAddress(address);
 
-      if (!traineeDAO.updateTrainee(trainee)) {
-        throw new NoSuchElementException("Trainee with ID " + id + " could not be updated.");
-      }
-      log.info("Trainee {} has been updated successfully.", id);
-      return true;
 
-    } catch (IllegalArgumentException e) {
-      log.error("Error updating trainee: {}", e.getMessage());
-      return false; // Возвращаем false или сообщение об ошибке
-    }
+    return traineeRepository.save(trainee);
   }
 
-  public boolean deleteTrainee(Long id) {
-    try {
-      if (!traineeDAO.deleteTrainee(id)) {
-        throw new NoSuchElementException("Trainee with ID " + id + " not found.");
-      }
-      log.info("Trainee {} has been deleted successfully.", id);
-      return true;
-    } catch (NoSuchElementException e) {
-      log.error("Error deleting trainee: {}", e.getMessage());
-      return false; // Возвращаем false или сообщение об ошибке
-    }
+
+  public List<Trainee> getAllTrainees() {
+    return traineeRepository.findAll();
   }
 
   public Trainee getTraineeById(Long id) {
-    Trainee trainee = traineeDAO.getTraineeById(id);
-    if (trainee == null) {
-      throw new NoSuchElementException("Trainee with ID " + id + " not found.");
+    Optional<Trainee> trainee = traineeRepository.findById(id);
+    if (trainee.isPresent()) {
+      return trainee.get();
+    } else {
+      throw new IllegalArgumentException("Trainee not found with id: " + id);
     }
```

Comment:
Good code, but what if you rewrite it like this. What do you think?
```
return traineeRepository.findById(id)
            .orElseThrow(()-> new IllegalArgumentException("Trainee not found with id: " + id));
```
Also what happens if id is empty?